src/app/editor/abstract-array-field/abstract-array-field.component.ts

```diff
@@ -0,0 +1,66 @@
+import { AbstractTrackerComponent } from '../abstract-tracker';
+
+import { EmptyValueService } from '../shared/services';
+
+/**
+ * Abstract component to share code of common operations of all array fields
+ * 
+ * Instance properties declared here only to resolve syntax errors. 
+ * Hence they need to be declared in children extending components (with decarators if necessary)
+ */
+export abstract class AbstractArrayFieldComponent extends AbstractTrackerComponent {
+
+  values: Array<any>;
+  schema: Object;
+  emptyValueService: EmptyValueService;
+  protected _emptyValue: any;
+
+  /**
+   * Called when a property of any element of the values is changed
+   * Used if values is a object array.
+   * 
+   * @param {any} event - new value
+   * @param {number} index - index of changed element in array
+   * @param {key} key - name of the changed property of the element in given index
+   * 
+   */
+  onValueChange(event: any, index: number, key: string) {
```

Comment:
I'd agree, but in some official examples. It is done like this and I followed that standard. In template where you specify the event binding, you write `(<event-name>)="onValueChange($event)`. Maybe they name this param `event` because of this.